File: sonet-kmp/shared/src/commonMain/kotlin/com/sonet/domain/model/Boo.kt
package com.sonet.domain.model
import kotlinx.serialization.Serializable
import kotlinx.datetime.Instant

@Serializable
data class Boo(
    val id: String,
    val userId: String,
    val username: String,
    val content: String,
    val createdAt: Instant,
    val expiresAt: Instant,
    val lifecycle: BooLifecycle = BooLifecycle.LIVE
)

enum class BooLifecycle { LIVE, GHOST, OBLIVION }

File: Sonet-iOS/Sonet-iOS/Domain/Entities/Boo.swift
import Foundation

struct Boo: Identifiable, Codable {
    let id: UUID
    let userId: UUID
    let username: String
    let userAvatar: URL?
    let content: String
    let mediaUrls: [URL]
    let createdAt: Date
    let expiresAt: Date
    let ghostsAt: Date
    let oblivionAt: Date
    var fireCount: Int
    var replyCount: Int
    var saveCount: Int
    var duetCount: Int
    var lifecycle: BooLifecycle
    
    var timeRemaining: TimeInterval {
        expiresAt.timeIntervalSinceNow
    }
    
    var isLive: Bool {
        lifecycle == .live
    }
    
    var isGhost: Bool {
        lifecycle == .ghost
    }
}

enum BooLifecycle: String, Codable {
    case live
    case ghost
    case oblivion
}
